ballerina/http_request.bal

```diff
@@ -512,13 +512,15 @@ public class Request {
 
     # Sets the request payload. This method overrides any existing content-type by passing the content-type
     # as an optional parameter. If the content type parameter is not provided then the default value derived
-    # from the payload will be used as content-type only when there are no existing content-type header.
+    # from the payload will be used as content-type only when there are no existing content-type header. If
+    # the payload is non-json typed value then the value is converted to json using the `toJson` method.
     #
-    # + payload - Payload can be of type `string`, `xml`, `json`, `byte[]`, `stream<byte[], io:Error?>`
+    # + payload - Payload can be of type `anydata`, `stream<byte[], io:Error?>`
```

Comment:
Would we have to explicitly mention xml, json, etc. as examples? Just thinking of someone getting started with Ballerina.

Reply:
We usually don't suggest users to use request/response objects with setPayload methods unless it is an advanced scenario like multipart. May be we can include a more detailed description in the spec